Sashanity/NoteTube
# File Upload and File Preview

Branch that adds upload and preview to the API. To use, run the server with firebase serve.

******
**Upload**

To test on Postman, first use the login API to get a token from firebase. Then, make sure to have these settings:

url: http://localhost:5000/notetube-f3f9c/us-central1/api/upload
type: POST

PARAMS:
-token: The auth token received after logging in.

BODY:
Make sure to set body to form-data
-file: The file to be uploaded.
-name: The name of the file
-subject: The subject that these notes were written for. Based on the list we decided upon.
-course: The class the notes were written for
-term: The semester these notes were written for
-instructor: The instructor of the class
-owner: The username of the user uploading the notes
-public: Whether the notes were public or not

RETURN (if successful):
-status: The status of whether the note was uploaded or not
-noteID: The ID of the note collection added to Firestore
-field: The data that was added to the collection:
--name: The name of the file
--filename: The name of the file in the storage bucket
--subject: The subject that these notes were written for. Based on the list we decided upon.
--course: The class the notes were written for
--term: The semester these notes were written for
--instructor: The instructor of the class
--owner: The username of the user uploading the notes
--public: Whether the notes were public or not
--uploader: The ID of the user that uploaded the note
--timestamp: The timestamp of when the note was uploaded
*******

*******
**Preview**

To test on Postman, follow these settings:

url: http://localhost:5000/notetube-f3f9c/us-central1/api/preview
type: GET

PARAMS:
-noteid: The ID of the note you want to preview
-token [OPTIONAL]: The token you get when you logged in

RETURN (if successful):
-Returns the file's contents in the response body. Will show the file as garbled hex in Postman, but if you copy+paste the URL into a browser, it will work. Note that it will be able to effectively preview .pdf files, but will prompt the user to download .doc/.docx files. 

NOTES:
-The token is an optional parameter. The token is used to verify the user if the user is trying to access their file that was set to private (public='false'). If the user is trying to access a private note when they are not authenticated or they are not the user that uploaded the note, it will return a 404 status code.
*******

EDIT: changed "class" to "course"
EDIT2: updated instructions to be more clear about previewing files on a browser
